src/theory/builtin/proof_checker.h

```diff
@@ -144,13 +140,21 @@ class BuiltinProofRuleChecker : public ProofRuleChecker
    */
   static void addMethodIds(std::vector<Node>& args, MethodId ids, MethodId idr);
 
+  /** get a TheoryId from a node, return false if we fail */
+  static bool getTheoryId(TNode n, TheoryId& tid);
```

Comment:
Could consider returning a pair instead.

Reply:
This is intentional actually, it makes the control flow simpler since you can have a condition that calls this method and then be ensured that under that condition tid is valid. This is a style we've been following for other utility methods.